components/Nav/MobileNav.tsx

```diff
@@ -0,0 +1,145 @@
+import { Disclosure, Transition } from "@headlessui/react";
+import { signIn } from "next-auth/react";
+import Link from "next/link";
+import { FunctionComponent } from "react";
+import { navigation, subNav, userSubNav } from "../../config/site_settings";
+import { UserNavigationItem } from "@/types/types";
+import { Session } from "next-auth";
+
+function classNames(...classes: string[]) {
+  return classes.filter(Boolean).join(" ");
+}
+
+interface MobileNavProps {
+  session: Session | null;
+  userNavigation: UserNavigationItem[];
+}
+
+const MobileNav: FunctionComponent<MobileNavProps> = ({
+  session,
+  userNavigation,
+}) => {
+  return (
+    <Transition
+      enter="transition-transform duration-200"
+      enterFrom="transform -translate-y-full opacity-0"
+      enterTo="transform translate-y-0 opacity-100"
+      className="absolute w-screen z-10 bg-neutral-100 dark:bg-black"
+    >
+      <Disclosure.Panel className="md:hidden relative z-10 border-b-2 border-black dark:border-white">
+        <div className="px-2 pt-2 pb-3 space-y-1">
+          {navigation.map((item) =>
+            item.href.includes("http") ? (
+              <Disclosure.Button
+                as="a"
+                key={item.name}
+                href={item.href}
+                target="_blank"
+                rel="noopener noreferrer"
+                className="nav-button block text-base font-medium"
+              >
+                {item.name}
+              </Disclosure.Button>
+            ) : (
+              <Link key={item.name} href={item.href}>
+                <Disclosure.Button
+                  as="div"
+                  className="nav-button block text-base font-medium"
+                >
+                  {item.name}
+                </Disclosure.Button>
+              </Link>
+            )
+          )}
+
+          <div className="pt-3 pb-3 border-t border-neutral-700 flex flex-col space-y-1">
+            {session ? (
+              <>
+                {userSubNav.map((item) => (
```

Comment:
same here: 

something like 
`<SubNav session={session} />`

Reply:
I can further refactor as you please. Will begin tomorrow 👍